compiler-cli/src/build_lock.rs

```diff
@@ -33,8 +44,50 @@ impl BuildLock {
 pub(crate) struct Guard(fslock::LockFile);
 
 #[test]
-fn locking() {
-    let lock = BuildLock::new().expect("make lock");
+fn locking_global() {
+    let lock = BuildLock::new_global().expect("make lock");
+    let _guard1 = lock.lock(&crate::telemetry::NullTelemetry);
+    println!("Locked!")
```

Comment:
Lol this test is a bit rubbish, why did I write it like this? Weird that it just prints. I should test to see if the lock can be aquired a second time by a different locker.

Reply:
Want me to do that or this a future issue?

Reply:
Up to you :)

Reply:
left as a future issue to limit scope of this PR: https://github.com/gleam-lang/gleam/issues/1936